drivers/hmis/app/models/hmis/hud/processors/geolocation_processor.rb

```diff
@@ -13,51 +13,81 @@ def schema
       nil
     end
 
-    def process(field, value)
-      return @processor.send(factory_name).destroy if value.nil? || value.empty?
+    def relation_name
+      :clh_location
+    end
 
+    def process(field, value)
       attribute_name = ar_attribute_name(field)
-
+      # The only expected attribute name is 'coordinates'
       raise ArgumentError, "Unexpected attribute for Geolocation: #{attribute_name}" unless attribute_name == 'coordinates'
 
-      attribute_value = JSON.parse(value, symbolize_names: true)
+      # if 'Geolocation.coordinates: nil' was submitted, destroy the clh location record
+      return destroy_record if value.nil? || value.empty?
+
+      attribute_value = clean_coordinate_value(value)
```

Comment:
exclude empty string values so we don't have to check for that case?
```suggestion
      attribute_value = clean_coordinate_value(value).compact_blank
```